File: app/src/main/java/com/example/mike/mp3player/client/MySearchSuggestionsProvider.java
package com.example.mike.mp3player.client;

import android.content.SearchRecentSuggestionsProvider;
import android.database.Cursor;
import android.net.Uri;

public class MySearchSuggestionsProvider extends SearchRecentSuggestionsProvider {

    public static final String AUTHORITY = "com.example.mike.mp3player.client.MySearchSuggestionsProvider";
    public static final int MODE = DATABASE_MODE_QUERIES;

    public MySearchSuggestionsProvider() {
        setupSuggestions(AUTHORITY, MODE);
    }

    @Override
    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs,
                        String sortOrder) {
        // gets recent suggestions
        Cursor recentSuggestions = super.query(uri, projection, selection, selectionArgs, sortOrder);

        // get string query
        String query = uri.getLastPathSegment().toLowerCase();
        return null;
    }
}

File: client/src/main/java/com/github/goldy1992/mp3player/client/MySearchSuggestionsProvider.kt
package com.github.goldy1992.mp3player.client

import android.content.SearchRecentSuggestionsProvider
import android.database.Cursor
import android.net.Uri

class MySearchSuggestionsProvider : SearchRecentSuggestionsProvider() {

    override fun query(uri: Uri,
                       projection: Array<String>,
                       selection: String,
                       selectionArgs: Array<String>,
                       sortOrder: String): Cursor? { // gets recent suggestions
        val recentSuggestions = super.query(uri, projection, selection, selectionArgs, sortOrder)
        // get string query
        val query = uri.lastPathSegment.toLowerCase()
        return null
    }

    companion object {
        const val AUTHORITY = "com.github.goldy1992.mp3player.client.MySearchSuggestionsProvider"
        const val MODE = DATABASE_MODE_QUERIES
    }

    init {
        setupSuggestions(AUTHORITY, MODE)
    }
}